src/routes/categories/containers/__tests__/CategoriesContainer.spec.js

```diff
@@ -0,0 +1,251 @@
+    path: rootCategory.path,
+    depth: 2,
+    language: language.code,
+    city: city.code,
+    allAvailableLanguages: new Map(languages.map(lng => [lng.code, `/${city.code}/${lng.code}`])),
+    models,
+    children
+  }
+
+  it('should display null if the route is not initialized', () => {
+    const state: StateType = prepareState()
+    const store = mockStore(state)
+    const navigation = createNavigationScreenPropMock()
+    navigation.state.key = 'route-id-0'
+    jest.doMock('../../../../modules/categories/components/Categories', () => MockCategories)
+    const CategoriesContainer = require('../CategoriesContainer').default
+
+    const result = TestRenderer.create(
+      <Provider store={store}><CategoriesContainer navigation={navigation} /></Provider>
+    )
+    expect(result.toJSON()).toBeNull()
+  })
+
+  const expectError = (state: StateType, message: string) => {
+    const store = mockStore(state)
+    const navigation = createNavigationScreenPropMock()
+    navigation.state.key = 'route-id-0'
+    jest.doMock('../../../../modules/categories/components/Categories', () => MockCategories)
+    const CategoriesContainer = require('../CategoriesContainer').default
+
+    const { getByText } = render(
+      <Provider store={store}><CategoriesContainer navigation={navigation} /></Provider>
+    )
+    expect(getByText(message)).toBeTruthy()
+  }
+
+  it('should display error if the route has the status error', () => {
+    const state: StateType = prepareState({
+      status: 'error',
+      path: rootCategory.path,
+      depth: 2,
+      language: language.code,
+      city: city.code,
+      message: 'Something went wrong with the route'
+    })
+    expectError(state, 'Something went wrong with the route')
+  })
+
+  it('should display error if cities could not be loaded', () => {
+    const state: StateType = prepareState(successfulRouteState, {
+      cities: {
+        status: 'error',
+        message: 'Something went wrong with the cities'
+      }
+    })
+    expectError(state, 'Something went wrong with the cities')
+  })
+
+  it('should display error if resourceCache could not be loaded', () => {
+    const state: StateType = prepareState(successfulRouteState, {
+      resourceCacheState: {
+        status: 'error',
+        message: 'Something went wrong with the resourceCache'
+      }
+    })
+    expectError(state, 'Something went wrong with the resourceCache')
+  })
+
+  const expectLoadingIndicator = (state: StateType) => {
+    const store = mockStore(state)
+    const navigation = createNavigationScreenPropMock()
+    navigation.state.key = 'route-id-0'
+    jest.doMock('../../../../modules/categories/components/Categories', () => MockCategories)
+    const CategoriesContainer = require('../CategoriesContainer').default
+    const result = TestRenderer.create(
+      <Provider store={store}><CategoriesContainer navigation={navigation} /></Provider>
+    )
+    jest.advanceTimersByTime(1000)
```

Comment:
It would be good to have this constant depend on the 800ms delay. At least this should be extracted to a constant.